aiohttp_sse/__init__.py

```diff
@@ -130,10 +132,16 @@ async def send(
 
         buffer.write(self._sep)
         try:
-            await self.write(buffer.getvalue().encode("utf-8"))
+            await asyncio.wait_for(  # TODO(PY311): Use asyncio.timeout
+                self.write(buffer.getvalue().encode("utf-8")),
```

Comment:
Hmm, does this actually block until the client acknowledges receipt?

Reply:
Here I just added a timeout, the rest of the behavior does not change.

Reply:
That uses curl, so maybe we can just do something similar with aiohttp?

I'm just looking back at the original issue though, I'm wondering if there's any better option, as the issue was described as "continued generating chunks to send on this connection, slowly saturating TCP buffers before finally simply hanging in the send call".

The solution only breaks from that hanging send call, so it's still filling the buffers with 400+ messages before it gets to that point. Not ideal...